torch/csrc/api/include/torch/nn/options/transformercoder.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <torch/arg.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/types.h>
+#include <torch/enum.h>
+
+#include <torch/nn/modules/transformerlayer.h>
+#include <torch/nn/modules/container/any.h>
+
+namespace torch {
+namespace nn {
+
+  /// Options for the `TransformerEncoder`
+  ///
+  /// Example:
+  /// ```
+  /// TransformerEncoderLayer encoderLayer(TransformerEncoderLayerOptions(512, 8).dropout(0.1));
+  /// auto options = TransformerEncoderOptions(encoderLayer, 6).norm(LayerNorm(LayerNormOptions({2})));
+  /// ```
+  struct TORCH_API TransformerEncoderOptions {
+    // This constructor will keep the a ref of passed in encoder_layer, so it keeps all the data in encoder_layer.
+    TransformerEncoderOptions(TransformerEncoderLayer encoder_layer, int64_t num_layers);
+    // This constructor will create a new TransformerEncoderLayer obj based on passed in encoder_layer_options.
+    TransformerEncoderOptions(const TransformerEncoderLayerOptions& encoder_layer_options, int64_t num_layers);
```

Comment:
Are there any other Options that hold modules? (So I can calibrate my expectations against those)

Reply:
:), no, this is the first time, a module need another module as an input parameter, I had the same question as you, and looked around for a while, did not find anyone. So we are on our own.